src/__tests__/realTimeReporter/utils.spec.ts

```diff
@@ -0,0 +1,176 @@
+
+    expect(startLaunchObject.attributes).toEqual([
+      { value: 'value', key: 'key' },
+      { value: 'mockValue', key: 'mockKey' }
+    ]);
+  });
+
+  test('should return startLaunchObject with system attributes in case of launch has not any attributes',
+    function () {
+      const startLaunchObject = getStartLaunchObj({});
+
+      expect(startLaunchObject.attributes).toEqual([
+        { value: 'mockValue', key: 'mockKey' }
+      ]);
+    }
+  );
+
+  test('should return startLaunchObject with system attributes in case of launch has empty attributes',
+    function () {
+      const startLaunchObject = getStartLaunchObj({ attributes: [] });
+
+      expect(startLaunchObject.attributes).toEqual([
+        { value: 'mockValue', key: 'mockKey' }
+      ]);
+    }
+  );
+});
+
+describe('publishEvent', function() {
+  test('should send event with message for current process', function () {
+    const spyProcessEmit = jest.spyOn(process, 'emit')
+      .mockImplementationOnce(function (event, msg) {
+        return {} as NodeJS.Process;
+      }
+    );
+
+    publishEvent('mockEvent', { data: 'mock' });
+
+    expect(spyProcessEmit).toHaveBeenCalledWith('mockEvent', { data: 'mock' });
+  });
+});
+
+
+describe('subscribeToEvent', function() {
+  test('should create listener on provided event for current process', function () {
+    const spyProcessOn = jest.spyOn(process, 'on')
+      .mockImplementationOnce(function (event, msg) {
+          return {} as NodeJS.Process;
+      }
+    );
+    const eventHandler = () => {};
+
+    subscribeToEvent('mockEvent', eventHandler);
+
+    expect(spyProcessOn).toHaveBeenCalledWith('mockEvent', eventHandler);
+  });
+});
+
+describe('getStack', function() {
+  test('should return array of NodeJS.CallSite for the function call', function () {
+    const stack = getStack();
+
+    expect(stack).toBeInstanceOf(Array);
+  });
+
+  test('the file name for the last item in stack should be equal to full this test file name', function () {
+    const stack = getStack();
+
+    expect(stack[0].getFileName()).toBe(__filename);
+  });
+});
+
+describe('getCaller', function() {
+  test('should return the caller\'s caller ', function () {
+    jest.spyOn(utils, 'getStack')
+      .mockReturnValueOnce([
+        {
+          getFileName: () => 'getStack call',
+        },
+        {
+          getFileName: () => 'getCaller call',
+        },
+        {
+          getFileName: () => 'getCaller caller call (this test)',
+        },
+        {
+          getFileName: () => 'caller\'s caller',
+        }
+      ]);
+    const caller = getCaller();
+
+    expect(caller.getFileName()).toBe('caller\'s caller');
+  });
+});
+
+describe('getCodeRef', function() {
+  test('invokes getCaller to get caller', function () {
+    const spyGetCaller = jest.spyOn(utils, 'getCaller');
+
+    getCodeRef('testItem');
+
+    expect(spyGetCaller).toBeCalledWith();
```

Comment:
I think in this case more suitable .toHaveBeenCalled() or just .toBeCalled()

Reply:
May be.
I used `toBeCalledWith` to focus that the function called withour any arguments)